### PR TITLE
Check data overage

### PR DESCRIPTION
**Pull request by johnworth**
Adds middleware to check for a data overage before allowing users to upload and save files through the DE.

### CODE REVIEW

File: src/terrain/routes/data_usage_api.clj [hunk 1]
```diff
@@ -10,17 +10,20 @@
...
+     (context "/data" []
+       (GET "/current" []
+         :middleware [require-authentication]
+         :summary schema/UserCurrentDataSummary
+         :description schema/UserCurrentDataDescription
+         :return schema/UserCurrentDataTotal
+         (ok (dua/user-current-usage (:username current-user))))
+
+       (GET "/overage" []
+         :middleware [require-authentication])))))
```
Comment (ianmcorvidae) [inline src/terrain/routes/data_usage_api.clj:27]: Not sure what's going on here? I'm honestly not sure what compojure will do with an empty route like this.

**Review by ianmcorvidae — APPROVED** [review]
One thing that possibly creeped in unintentionally, but the main stuff of the PR (middleware, and adding the middleware to stuff) looks good to me

File: src/terrain/routes/data_usage_api.clj (hunk 1 above)
Comment (johnworth) [reply src/terrain/routes/data_usage_api.clj]: I decided to remove it. It was originally going to be in there for debugging purposes, but it doesn't seem necessary.

Comment (johnworth) [reply src/terrain/routes/data_usage_api.clj]: Also, thanks for catching that.

**Pull request merged**